timemachine/fe/plots.py

```diff
@@ -454,3 +454,208 @@ def plot_as_png_fxn(f, *args, **kwargs) -> bytes:
+        chiral_atom_idxs = sys.chiral_atom.potential.idxs
+        chiral_atom_params = sys.chiral_atom.params
+        keep_idxs = []
+        for b_idx, idxs in enumerate(chiral_atom_idxs):
+            if filter_fn(idxs):
+                keep_idxs.append(b_idx)
+        chiral_atom_ks.append(chiral_atom_params[keep_idxs])
+
+    chiral_atom_idxs = chiral_atom_idxs[keep_idxs]
+    chiral_atom_ks = np.array(chiral_atom_ks).T  # chiral_atoms x K
+    num_chiral_atoms = chiral_atom_ks.shape[0]
+
+    for b_idx in range(num_chiral_atoms):
+        if chiral_atom_ks[b_idx][0] != chiral_atom_ks[b_idx][-1]:
+            linestyle = "dashed"
+            label = f"{chiral_atom_idxs[b_idx]}"
+            alpha = 1.0
+        else:
+            linestyle = "dotted"
+            label = None
+            alpha = 0.1
+
+        axs[row, 0].plot(xs, chiral_atom_ks[b_idx], linestyle=linestyle, label=label, alpha=alpha)
+
+    axs[row, 0].set_title("chiral_atom force constants")
+    axs[row, 0].set_ylabel("force constant")
+    axs[row, 0].set_ylim(-100, 1200)
+    axs[row, 0].set_xlabel("lambda window")
+    # axs[row, 0].legend()
+
+
+def _plot_torsion_interpolation(xs, systems, filter_fn, axs, row):
+    torsion_ks = []  # K x # torsions
+    torsion_bs = []  # K x # torsions
+    for sys in systems:
+        torsion_idxs = sys.torsion.potential.idxs
+        torsion_params = sys.torsion.params
+        keep_idxs = []
+        for b_idx, idxs in enumerate(torsion_idxs):
+            if filter_fn(idxs):
+                keep_idxs.append(b_idx)
+        torsion_ks.append(torsion_params[keep_idxs, 0])
+        torsion_bs.append(torsion_params[keep_idxs, 1])
+
+    torsion_idxs = torsion_idxs[keep_idxs]
+    torsion_ks = np.array(torsion_ks).T  # torsions x K
+    torsion_bs = np.array(torsion_bs).T * (180 / np.pi)  # torsions x K *
+    num_torsions = torsion_ks.shape[0]
+
+    for b_idx in range(num_torsions):
+        if torsion_ks[b_idx][0] != torsion_ks[b_idx][-1]:
+            linestyle = "solid"
+            label = f"{torsion_idxs[b_idx]}"
+            alpha = 1.0
+        else:
+            linestyle = "dotted"
+            label = None
+            alpha = 0.1
+
+        axs[row, 0].plot(xs, torsion_ks[b_idx], linestyle=linestyle, label=label, alpha=alpha)
+        axs[row, 1].plot(xs, torsion_bs[b_idx], linestyle=linestyle, label=label, alpha=alpha)
+
+    axs[row, 0].set_title("torsion force constants")
+    axs[row, 1].set_title("torsion degrees")
+
+    axs[row, 0].set_ylabel("force constant")
+    axs[row, 1].set_ylabel("torsion degree")
+
+    axs[row, 0].set_xlabel("lambda window")
+    axs[row, 1].set_xlabel("lambda window")
+
+    # axs[row, 0].legend(bbox_to_anchor=(1, 0.5))
+    # axs[row, 1].legend(bbox_to_anchor=(1, 0.5))
+
+
+def plot_interpolation_schedule(st, filter_fn, fig_title):
+    fig, axs = plt.subplots(5, 2, figsize=(9, 12))
+    # plot the force constant and equilibrium bond lengths along lambda
+    n_windows = 21
```

Comment:
If this is slow, also note that `setup_intermediate_state` can be called inside of a function that takes a lambda and returns parameter arrays, and that function should be jax-transformable (`vmap`, `jit`, etc.). This can make a big difference in my experience.

Reply:
willfix to higher resolution and use jit

Reply:
(running into some issues with jit right now, namely the returned VacuumSystems are not jit-compatible, so I will need to figure out how to use Matt's approach later)

Reply:
(discussed offline with @mcwitt  and will defer jit to a later pass, for now, setting n_windows=48 to improve resolution in the near term).